File: backend/TrackRanker.Api/Models/TrainingSession.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace TrackRanker.Api.Models;

public sealed class TrainingSession
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string Id { get; set; } = string.Empty;

    public required string Title { get; set; }

    [BsonRepresentation(BsonType.String)]
    public SessionType SessionType { get; set; }

    [BsonDateTimeOptions(Kind = DateTimeKind.Utc)]
    public DateTime SessionDate { get; set; }

    public required string Prescription { get; set; }

    public string? Purpose { get; set; }

    public string? FocusCue { get; set; }

    public string? SuccessCriteria { get; set; }

    public int? IntendedIntensity { get; set; }

    public string? CoachNotes { get; set; }

    [BsonRepresentation(BsonType.String)]
    public TrainingSessionStatus Status { get; set; }

    [BsonDateTimeOptions(Kind = DateTimeKind.Utc)]
    public DateTime CreatedAtUtc { get; set; }

    [BsonDateTimeOptions(Kind = DateTimeKind.Utc)]
    public DateTime UpdatedAtUtc { get; set; }
}

File: frontend/src/types/trainingSession.ts
export const sessionTypes = [
  "Acceleration",
  "MaxVelocity",
  "SpeedEndurance",
  "SpecialEndurance",
  "Tempo",
  "Starts",
  "Competition",
  "Recovery",
  "Other",
] as const;

export const sessionStatuses = ["Planned", "Completed", "Cancelled"] as const;

export type SessionType = (typeof sessionTypes)[number];
export type TrainingSessionStatus = (typeof sessionStatuses)[number];

export interface TrainingSession {
  id: string;
  title: string;
  sessionType: SessionType;
  sessionDate: string;
  prescription: string;
  purpose: string | null;
  focusCue: string | null;
  successCriteria: string | null;
  intendedIntensity: number | null;
  coachNotes: string | null;
  status: TrainingSessionStatus;
  createdAtUtc: string;
  updatedAtUtc: string;
}

export interface TrainingSessionInput {
  title: string;
  sessionType: SessionType;
  sessionDate: string;
  prescription: string;
  purpose: string | null;
  focusCue: string | null;
  successCriteria: string | null;
  intendedIntensity: number | null;
  coachNotes: string | null;
  status: TrainingSessionStatus;
}
